### PR TITLE
Cleaner output of docker image building scripts

### PR DESCRIPTION
**Pull request by potiuk**
There was some "junk" output generated by the scripts that are
used in Airflow image building. The junk has been cleaned up so
that no unnecessary warnings are generated.

Separated out from #20238

Based on #20678  - so please take a look at the last commit only.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: scripts/docker/compile_www_assets.sh [hunk 1]
```diff
@@ -35,8 +34,27 @@ function compile_www_assets() {
...
+    yarn run prod 2>/tmp/out-yarn-run.txt
+    res=$?
+    if [[ ${res} != 0 ]]; then
+        >&2 echo
+        >&2 echo "Error when running yarn install:"
+        >&2 echo
+        >&2 cat /tmp/out-yarn-run.txt
+        exit 1
+    fi
+    rm -f /tm/out-yarn-run.txt
```
Comment (ashb) [inline scripts/docker/compile_www_assets.sh]:
```suggestion
    rm -f /tmp/out-yarn-run.txt
```

File: scripts/docker/compile_www_assets.sh [hunk 2]
```diff
@@ -35,8 +34,27 @@ function compile_www_assets() {
...
-    yarn install --frozen-lockfile --no-cache
-    yarn run prod
+    set +e
+    yarn install --frozen-lockfile --no-cache 2>/tmp/out-yarn-install.txt
+    local res=$?
+    if [[ ${res} != 0 ]]; then
+        >&2 echo
+        >&2 echo "Error when running yarn install:"
+        >&2 echo
+        >&2 cat /tmp/out-yarn-install.txt
```
Comment (ashb) [inline scripts/docker/compile_www_assets.sh]: File left around on error exit - does it matter?

**Review by ashb — APPROVED** [review]
Minor comment, then this second commit LGTM.

File: scripts/docker/compile_www_assets.sh (hunk 1 above)
Comment (potiuk) [reply scripts/docker/compile_www_assets.sh]: +1

File: scripts/docker/compile_www_assets.sh (hunk 2 above)
Comment (potiuk) [reply scripts/docker/compile_www_assets.sh]: Not much, but I will remove it.

**Review by uranusjr — APPROVED** [review]

**Pull request merged**